pagerduty.tf

```diff
@@ -1,24 +1,118 @@
-resource "newrelic_alert_channel" "pagerduty" {
+locals {
+  summary       = "{{ annotations.title.[0] }}"
+  customDetails = <<-EOT
+    {
+      "condition_name": {{json accumulations.conditionName.[0]}},
+      "details": {{json issueTitle}},
+      "duration": {{#if issueDurationMs}}{{issueDurationMs}}{{else}}0{{/if}},
+      "event_type": "INCIDENT",
+      "incident_acknowledge_url": {{json issueAckUrl}},
+      "incident_url": {{json issuePageUrl}},
+      "policy_name": {{json accumulations.policyName.[0]}},
+      {{#if accumulations.runbookUrl}}"runbook_url": {{json accumulations.runbookUrl.[0]}},{{/if}}
+      "severity": "{{#eq 'HIGH' priority}}WARNING{{else}}{{priority}}{{/eq}}",
+      "status": {{json status}},
+      "timestamp": {{updatedAt}},
+      "timestamp_utc_string": {{json issueUpdatedAt}}
+    }
+    EOT
+}
+
+resource "newrelic_notification_destination" "pagerduty" {
   count = var.enable_pagerduty_notifications ? 1 : 0
 
-  name = var.newrelic_app_name
-  type = "pagerduty"
+  account_id = var.newrelic_account_id
+  name       = var.newrelic_app_name
+  type       = "PAGERDUTY_SERVICE_INTEGRATION"
 
-  config {
-    service_key = var.pagerduty_service_key
+
+  auth_token {
```

Comment:
There is an example here: https://registry.terraform.io/providers/newrelic/newrelic/3.35.0/docs/resources/notification_destination#additional-examples

At the beginning the `property` section did not exist, but the provider throwed an error because at least one was necessary. So I added that one (I set it to `false` as the rest of the services we have this property is also `false`).

Looking at the examples provided in the link above we could do just, but I don't like it very much:
```
  property {
    key = ""
    value = ""
  }
```

`two_way_integration` documentation: https://docs.newrelic.com/docs/alerts-applied-intelligence/notifications/notification-integrations/#two-way-pagerduty